##What if we use that the former inmate was or not was not ever employed in a CEO transitional job (third row on the table):

##P0=0.637, P1=0.69



In [6]:
import os
import sys
import numpy as np
import pylab as pl
%pylab inline


Populating the interactive namespace from numpy and matplotlib


In [7]:
alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=63.7*0.01 
P_1=69*0.01

n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1

In [8]:
#define the sample proportion first
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.667721479959


In [9]:
# i am goonna create a little one line function to calculate the standard dev, it is not really needed, but just to show you how you do such a thing
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_2y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.637 409 564 0.0305917288695


In [10]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y

1.73249443424


In [11]:
##p-value for employment after 2 years: since the largest number we read off the table for alpha=0.05 is smaller then the value for our statistic our p-value will be smaller then calculated using .9998
##thus using 0.9998 is a **conservative** approach. 

p_2y=1-0.9998


def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


##Chisquare: lets see what the chi sq statistics says about the employment for >6 consecutive months (raw 6)
                       | employed in subsadized job | employed  | not employes |     |
                       |----------------------------|-----------|--------------|-----|
                       | test sample                | 14.7*5.64 | 85.3*5.64    | 564 |
                       | control sample             | 11.9*4.09 | 88.1*4.09    | 409 |
                       | total                      | 131.579   | 841.421      | 973 |
                      

In [13]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


Because 1.589 is smaller than 3.84 that is the chi-square value at the one df with alpha equal 0.05. Base on this result, we cannot reject the null hypothesis.